src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -46,6 +46,22 @@ public class ObservableCollection<T> : Collection<T>, INotifyCollectionChanged,
         /// <exception cref="ArgumentNullException"> collection is a null reference </exception>
         public ObservableCollection(IEnumerable<T> collection) : base(CreateCopy(collection)) { }
 
+        /// <summary>
+        /// Initializes a new instance of the ObservableCollection class
+        /// that contains elements copied from the specified list
+        /// </summary>
+        /// <param name="list">The list whose elements are copied to the new list.</param>
+        /// <remarks>
+        /// The elements are copied onto the ObservableCollection in the
+        /// same order they are read by the enumerator of the list.
+        /// </remarks>
+        /// <exception cref="ArgumentNullException"> list is a null reference </exception>
+        public ObservableCollection(List<T> list)
+            : base((list != null) ? new List<T>(list.Count) : list)
+        {
+            CreateCopy(list);
+        }
```

Comment:
I agree that the test should've failed, but the tests Are running against the new surface area so I guess I'm not sure why the test is not failing. I'll debug to take a look.